### PR TITLE
Bugfix/openmp

### PR DESCRIPTION
**Pull request by arbennett**
This solves #355 and also includes a minor fix for edge cases with no time zone information in the time string. Tested with gfortran, passes all tests and is bit for bit with previous versions.

### CODE REVIEW

File: build/source/driver/summa_modelRun.f90 [hunk 1]
```diff
@@ -185,51 +185,65 @@ subroutine summa_runPhysics(modelTimeStep, summa1_struc, err, message)
...
-
- nThreads = 1
- !$ nThreads = omp_get_num_threads()
-
- ! use dynamic scheduling with chunk size of one:
- !  -- new chunks are assigned to threads when they become available
- !  -- start with the more expensive GRUs, and add the less expensive GRUs as threads come available
-
- !$omp do schedule(dynamic, 1)   ! chunk size of 1
+   !!$omp parallel
```
Comment (bartnijssen) [inline build/source/driver/summa_modelRun.f90]: delete

File: build/source/driver/summa_modelRun.f90 [hunk 2]
```diff
@@ -185,51 +185,65 @@ subroutine summa_runPhysics(modelTimeStep, summa1_struc, err, message)
...
+  !$omp parallel  default(none) &
+  !$omp          private(iGRU, jGRU) &  ! GRU indices are private for a given thread
+  !$omp          shared(openMPstart, openMPend)   & ! access constant variables
+  !$omp          shared(timeGRUstart, timeGRUcompleted, timeGRU, ixExpense, kGRU)  & ! time variables shared
+  !$omp          shared(summa1_struc, gru_struc) &
+  !$omp          private(err, cmessage)
+ ! nThreads = 1
```
Comment (bartnijssen) [inline build/source/driver/summa_modelRun.f90]: delete or uncomment

File: build/source/driver/summa_modelRun.f90 [hunk 3]
```diff
@@ -185,51 +185,65 @@ subroutine summa_runPhysics(modelTimeStep, summa1_struc, err, message)
...
+ ! !$ nThreads = omp_get_num_threads()
```
Comment (bartnijssen) [inline build/source/driver/summa_modelRun.f90]: remove commented out code

File: build/source/engine/time_utils.f90 [hunk 4]
```diff
@@ -125,6 +125,7 @@ subroutine extractTime(refdate,iyyy,im,id,ih,imin,dsec,ih_tz,imin_tz,dsec_tz,err
 
  ! FIELD 3: Advance to the ih_tz:imin_tz string
  istart=nsub+1
+ if(istart>n) return  ! no more time info
```
Comment (bartnijssen) [inline build/source/engine/time_utils.f90]: Make separate PR for this

**Comment by bartnijssen** [comment]
@arbennett : If you made changes in response to the last review comments can you let me know what you did (just one word/ responses to each of the questions). That way I know that something happened and whether it is ready for re-review (and merge).
